libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -68,8 +68,14 @@ void StaticAnalyzer::endVisit(FunctionDefinition const&)
 	m_nonPayablePublic = false;
 	m_constructor = false;
 	for (auto const& var: m_localVarUseCount)
-		if (var.second == 0)
-			m_errorReporter.warning(var.first->location(), "Unused local variable");
+		if (var.second == 0 && var.first->isCallableParameter())
+			m_errorReporter.warning(
+				var.first->location(), 
+				"Unused local variable. Remove or comment out the variable name to silence this warning");
+		else if (var.second == 0) 
```

Comment:
This condition is duplicated above. Use nesting if necessary to avoid this duplication.